### PR TITLE
8331030: langtools/tools/javac/tree tests fail with SOE with fastdebug and -Xcomp

### PR DESCRIPTION
**Pull request by sormuras**
Please review this change that now uses a source generator and compiler task to verify no SOE is thrown.

This change replaces over 1000 lines of code that were being compiled by running this test as well as other tests crawling over test files. Some of those crawling tests failed to parse this formerly large file. The new size of this test file should be no problem for them.

The level of nesting is increased from 567 to 1000.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331030](https://bugs.openjdk.org/browse/JDK-8331030): langtools/tools/javac/tree tests fail with SOE with fastdebug and -Xcomp (**Bug** - P4)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [6d51348f](https://git.openjdk.org/jdk/pull/18939/files/6d51348f5edfa1586ae3255539bdb1a0f156d3ca)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18939/head:pull/18939` \
`$ git checkout pull/18939`

Update a local copy of the PR: \
`$ git checkout pull/18939` \
`$ git pull https://git.openjdk.org/jdk.git pull/18939/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18939`

View PR using the GUI difftool: \
`$ git pr show -t 18939`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18939.diff">https://git.openjdk.org/jdk/pull/18939.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18939#issuecomment-2075422551)

### CODE REVIEW

File: test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java [hunk 1]
```diff
@@ -23,1165 +23,37 @@
...
-        }
-        }
-        }
-        }
-        }
+    public static void main(String... args) {
+        var lines = new ArrayList<String>();
+        lines.add("class Test {");
+        lines.add("  static { ");
+        for (int i = 0; i < 1000; i++) lines.add("    synchronized (Test.class) {");
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java:42]: Suggest using a  named constant  for the two values of 1000

Comment (sormuras) [reply test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java]: Done via https://github.com/openjdk/jdk/pull/18939/commits/6d51348f5edfa1586ae3255539bdb1a0f156d3ca

**Review by jonathan-gibbons — APPROVED** [review]

File: test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java [hunk 2]
```diff
@@ -23,1165 +23,39 @@
 
 /**
  * @test
- * @bug 8322992
+ * @bug 8322992 8331030
  * @summary Javac fails with StackOverflowError when compiling deeply nested synchronized blocks
  * @compile SOEDeeplyNestedBlocksTest.java
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java:28]: This line is not necessary; the file will be compiled if needed by the `@run main`

Comment (sormuras) [inline test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java:28]:
```suggestion
```

Auto-suggestion.

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

File: test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java (hunk 2 above)
Comment (sormuras) [reply test/langtools/tools/javac/patterns/SOEDeeplyNestedBlocksTest.java]: Line removed.

**Pull request closed without merging**